src/components/shared/NeedSignIn.vue

```diff
@@ -15,21 +15,17 @@
     </div>
 </template>
 
-<script>
-import { mapState } from 'pinia'
+<script setup lang="ts">
 import { useUserStore } from '../../stores/user'
 import TheButton from './TheButton.vue'
 
-export default {
-  name: 'NeedSignIn',
-  props: ['text'],
-  components: {
-    TheButton
-  },
-  computed: {
-    ...mapState(useUserStore, ['isSignedIn'])
+const userStore = useUserStore()
+
+defineProps({
+  text: {
+    type: String
   }
-}
+})
```

Comment:
it's better to use the TS syntactic sugar for props definition:
```ts
defineProps<{text: string;}>()

Reply:
Had no previous knowledge about this syntax, **corrected**.